test/CMakeLists.txt

```diff
@@ -43,7 +44,10 @@ include(Fixtures.cmake)
 
 foreach(test_target ${tests})
   add_executable(${test_target} "${test_target}.cpp")
-  gtest_discover_tests(${test_target})
+  gtest_discover_tests(
+    ${test_target}
+    TEST_FILTER -Timedemo.*
+  )

```

Comment:
I'm a bit split on this one. Why would we not want to run this and get coverage from it? Are the drawbacks significant?

How that DRLG and Item RNG is covered I think it might be ok to have end-to-end tests for covering the majority of the remaining code.

Reply:
Cause we can't currently. 😢 
For a running game we need more data from the mpq.
For example `LoadLevelSOLData` or dun-files.
Also the trick with providing `pMegaTiles` doesn't work well with `LoadLvlGFX`.
But perhaps i miss something. 🙂

Reply:
Yeah just realized that 😅

But we could make a recording of a shareware run and that would still cover a lot of the mechanics 🤔

Reply:
We could download the `spawn.mpq` from `devilutionx-assets` for the tests.
But I don't know how to add this to (correctly) CMake.
Can someone help with this (for example add a commit to this pr) or should we leave this to a follow-up pr?
If wanted I can record the two small tests again with the spawn version,

Reply:
If you record the demo I will add it, but probably as part of the GitHub action script rather then CMake :)

Reply:
But this method would only work when running the game in ci or?
If the repo is cloned and the tests are run locally it will fail or?

Reply:
The readme clearly states that you must have diabdat.mpq or spawn.mpq for things to work. So that is the humans respectability.

Reply:
The readme also says you can provide them with `--data-dir`, but this doesn't work for unit tests or?

Reply:
Im not sure i follow your point?

Reply:
Sorry if I was unclear.

Following scenario
- New Developer
- Clones repro
- Compiles program
- Mpq's are not present in default search directory
- Developer uses `--data-dir` to pass a directory where the mpq's stored
- Works 👍 
- Tries to run unit tests
- Unit tests can't find mpq's
- Developer can't use `--data-dir` to pass the directory where the mpq's are stored
- Bad 😢 
- Developer copy's mpq's to bin directory manually, cause this is the only way to fix it. Or is there another way?

I thought we could make the work for the developer more easy if we always download `spawn.mpq`.

Side-node:
We could also ensure that we only load `spawn.mpq` when running in `HeadlessMode` that would make working locally consistent with github actions.

Reply:
It wasn't so much the steps i didn't get as why it was important.

> Developer can't use --data-dir to pass the directory where the mpq's are stored

If you think this is relevant then just make the additional calls for setting this up during the test would solve it. I don't think it's important though. It's ok to expect developers (who are skilled enough to make use of tests) to also be able to put a file in a specific place. We could add it to the build readme "For tests mpq files must be in folder X" if you think it's important.

> I thought we could make the work for the developer more easy if we always download spawn.mpq.

But some of the tests wont work without diabdat.mpq anyway so i just don't see the point here.

They also already seam upset that we will download ZT dependencies by default, adding another 25MB download rather then asking them to move a file that they are already required to have probably won't make them much happier.

>  We could also ensure that we only load spawn.mpq when running in HeadlessMode that would make working locally consistent with github actions.

I don't think it's good to bind these two options to each other. I'm not asking for the current demos to be removed, I just would like it if there is at least one demo that can be run on the CI so that we can have automatic benefits from the work done here :)

Reply:
I changed the following
- Recorded demos with spawn version.
- Removed filter for timedemo unit tests. Now they should also run in github actions.
- Added an assert and comment that spawn.mpq or diabdat.mpq needs to be present for the tests to run

Now the github actions need to be adjusted to download spawn.mpq.

Hope this helps to get a better code coverage. 🙂

Reply:
This is so exciting 😀